File: Creational/Abstract Factory/Java/ConcreteFactory2.java

public class ConcreteFactory2 implements AbstractFactory {

	@Override
	public AbstractProductA createProductA() {
		return new ConcreteProductA_2();
	}

	@Override
	public AbstractProductB createProductB() {
		return new ConcreteProductB_2();
	}

	@Override
	public AbstractProductC createProductC() {
		return new ConcreteProductC_2();
	}
	
}

File: Creational/Abstract Factory/Kotlin/ConcreteFactory2.kt
class ConcreteFactory2: AbstractFactory {

    override fun createProductA(): AbstractProductA {
        return ConcreteProductA2()
    }

    override fun createProductB(): AbstractProductB {
        return ConcreteProductB2()
    }

    override fun createProductC(): AbstractProductC {
        return ConcreteProductC2()
    }

}